examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,396 @@
+def get_results_mice_imputation_includingy(X_incomplete, y, num_imputations=5):
+    Xy = np.column_stack((X_incomplete, y))
+    Qbar, T = get_results_mice_imputation(Xy, y, num_imputations)
+    return Qbar, T
+
+
+# Now lets run all these imputation procedures.
+# We use the Boston dataset and analyze the outcomes of the beta coefficients
+# and their standard errors. We standardize the data before running the
+# procedure to be able to compare the coefficients. We run the procedure for
+# MCAR missingness only.
+#
+# Loading the data
+dataset = load_boston()
+X_full, y = dataset.data, dataset.target
+
+# Standardizing the data
+scaler = StandardScaler()
+X_scaled = scaler.fit_transform(X_full)
+y_scaled = stats.zscore(y)
+
+# Start the procedure
+print("Executing Example 1 MCAR Missingness...")
+
+# First, make the data incomplete with a MCAR mechanism.
+X_incomplete = ampute(X_scaled, mech="MCAR")
+
+# Second, run all the imputation procedures as described above.
+full_coefs, full_vars = get_results_full_dataset(X_scaled, y_scaled)
+chained_coefs, chained_vars = get_results_chained_imputation(
+        X_incomplete, y_scaled)
+mice_coefs, mice_vars = get_results_mice_imputation(
+        X_incomplete, y_scaled)
+mice_y_coefs, mice_y_vars = get_results_mice_imputation_includingy(
+        X_incomplete, y_scaled)
+
+# Combine the results from the four imputation procedures.
+coefs = [full_coefs, chained_coefs, mice_coefs, mice_y_coefs]
+vars = [full_vars, chained_vars, mice_vars, mice_y_vars]
+errorbars = [1.96 * np.sqrt(v) for v in vars]
+
+# And plot the results
+n_situations = 4
+n = np.arange(n_situations)
+n_labels = ['Full Data', 'IterativeImputer',
+            'Mice Imputer', 'Mice Imputer with y']
+colors = ['r', 'orange', 'b', 'purple']
+width = 0.3
+plt.figure(figsize=(24, 32))
+
+plt1 = plt.subplot(211)
+for j in n:
+    plt1.bar(np.arange(len(coefs[j])) + (3*j*(width/n_situations)),
+             coefs[j], width=width, color=colors[j])
+plt.legend(n_labels)
+
+plt2 = plt.subplot(212)
+for j in n:
+    plt2.bar(np.arange(len(errorbars[j])) + (3*j*(width/n_situations)),
+             errorbars[j], width=width, color=colors[j])
+
+plt1.set_title("MCAR Missingness")
+plt1.set_ylabel("Beta Coefficients")
+plt2.set_ylabel("Standard Errors")
+plt1.set_xlabel("Features")
+plt2.set_xlabel("Features")
+plt.show()
+
+
+###############################################################################
+# EXAMPLE 2. SHOW MULTIPLE IMPUTATION IN A PREDICTION CONTEXT.
+
+
+# In this example, we show how to apply multiple imputation in a train/test
+# situation. There are two approaches to get the end result of the prediction
+# model. In approach 1 you calculate the evaluation metric for every i in m and
+# later average these values. In approach 2 you average the predictions of
+# every i in m and then calculate the evaluation metric. We test both
+# approaches.
+
+# Apply the regression model on the full dataset as a way of comparison.
```

Comment:
I think that we need to reduce the number of functions. I don't think that we actually need any function if we use the pipeline and set the random state whenever this is necessary.

Reply:
Sorry, how broadly does this apply? What does "any function" refer to?